woocommerce/woocommerce-ios
# [Mobile Payments] Modal view: Add support for device rotation

Closes #4094 

⚠️ This PR is against the feature branch implementing support for Card Present Payments, not against develop ⚠️
⚠️ Testing this PR requires a hardware reader ⚠️

| Portrait | Landscape |
| ---- | ---- |
| <img src="https://user-images.githubusercontent.com/2722505/117060622-19815680-acef-11eb-9981-e89552c71afb.PNG" width="350"/> | <img src="https://user-images.githubusercontent.com/2722505/117060621-18e8c000-acef-11eb-8bdf-ae0debd4356c.PNG" width="350"/> |
| <img src="https://user-images.githubusercontent.com/2722505/117060626-1a19ed00-acef-11eb-8be4-e17450f2220c.PNG" width="350"/> | <img src="https://user-images.githubusercontent.com/2722505/117060624-19815680-acef-11eb-9b10-3997c9d043d2.PNG" width="350"/> |

## Changes
* Switch the axis of the main stack view to horizontal when the device is rotated.
* Adjust a couple of constraints in the xib to get what we want.

## How to test
Testing requires:

1. Setting up a store for Card Present Payment Testing (also P91TBi-4BH-p2 for more specific instructions)
2. An external hardware reader.

* Checkout the branch, run bundle exec pod install, just for kicks and giggles.
* Build and run on a device.
* Switch an external card reader on. The only card reader supported by now is the BBPOS Chipper 2X BT.
* Log in to an account that matches a store that has been configured for testing.
* Navigate to Settings in the app. (Tap the gear button in the top right)
* In the "Store Settings" section, tap "Manage Card Reader".
* Make sure the card reader is on and the blue light is blinking (we don't do any error handling yet, so it's important that things are setup for the happy path)
* Tap "Connect card reader"
* Wait until the app suggests one reader to connect to.
* Tap "Connect to reader"
* Make sure the reader light turns solid blue.
* Navigate to an order eligible for Card Present Payments (it has to be a store enrolled to WCPay, and the order needs to have as payment method associated Cash on Delivery or none).
* Rotate the device
* Tap the "Collect Payment" button.
* Notice the alert. Rotate the device a couple of times, notice the changes in the alert.
* Tap the card to the reader when requested.
* If everything goes well, the card should be charged, and the payment should be completed.
* Rotate the device a couple of times as well. 
* For extra fun, repeat the flow (collecting a payment) with the device in portrait and landscape orientations



Update release notes:
